src/Valit/Rules/ValitRuleExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using System.Linq;
+using Valit.Exceptions;
+using Valit.Result;
+
+namespace Valit.Rules
+{
+    internal static  class ValitRuleExtensions
+    {
+        internal static IValitResult ValidateRules<TObject>(this IEnumerable<IValitRule<TObject>> rules, IValitStrategy strategy, TObject @object) where TObject : class
+        {
+            rules.ThrowIfNull();
+
+            var result = ValitResult.Success;
+
+            foreach(var rule in rules.ToList())
+            {
+                result &= rule.Validate(@object);
+
+                if(!result.Succeeded)
+                {
+                    strategy.Fail(rule, result, out bool cancel);
```

Comment:
I think I said this a couple of days ago, but maybe lets make `strategy.Fail` return a `bool`?

Reply:
@arekbal did this as an out param. I think he's the right person to ask why we actualy use it :D

Reply:
Cleaner API when implementing that method. Instead of guessing or going through some docs to find why method returns bool and what it means(true for cancel or continue?), one just implements a method with clear declaration.